### ISSUE TITLE
Arm64 Image

### ISSUE THREAD

**Issue by m4rcster** [calcom/docker#81]
Hi there,

Is there a plan to provide an arm64 image? 

https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/

**Comment by AmoghSaxena** [comment]
Hey! @m4rcster I was trying the same on my RaspberryPi will update you if I find any progress.

**Comment by finaldzn** [comment]
Did you try to build with the --platform key ?

**Comment by AmoghSaxena** [comment]
> Did you try to build with the --platform key ?

Not yet... But can try if you guide me.

**Comment by DrMxrcy** [comment]
Gonna have to +1 on this. Running this on ARM would be a game changer.

**Comment by krumware** [comment]
I'll give this a go on a test branch in the morning as I also look into the GitHub actions memory issues 👍👍

**Comment by DrMxrcy** [comment]
> I'll give this a go on a test branch in the morning as I also look into the GitHub actions memory issues 👍👍
> 
> (there were platform/buildx issues with GHA last time I gave this a go specifically and had to move on to other things)

Thanks!

**Comment by DrMxrcy** [comment]
Any updates on this?

**Comment by krumware** [comment]
In progress! Should have a good update early this coming week

**Comment by DrMxrcy** [comment]
> In progress! Should have a good update early this coming week

Let's go! Thanks!

**Comment by krumware** [comment]
https://github.com/calcom/docker/pull/144

Working through testing the github actions. `calendso/calendso:canary` will be the initial image tag for testing on the rpi.
There will be an initial restriction around the web_app_url where localhost will appear for the event URL paths, but we'll work through that.

**Comment by DrMxrcy** [comment]
Thanks so much on the updates! We will be testing it!

**Comment by DrMxrcy** [comment]
Perfect!

**Comment by krumware** [comment]
@erulabs & @DrMxrcy 
https://hub.docker.com/r/calendso/calendso/tags `canary` is ready for you to test

Again, just noting: There will be an initial restriction around the web_app_url where localhost will appear for the event URL paths, but we'll work through that.

Thanks for your help!

**Comment by erulabs** [comment]
Hey @krumware - thank you so much! We're looking forward to hosting this for our own uses over at https://kubesail.com - I'm also going to make an installer for Kubernetes if that's alright with you! The main goal of getting ARM64 support was also so that it could run on our little home-hosting computer, the https://pibox.io 👯 We'd love to chat with you about partnering up if you're interested - otherwise we're happy to help generate new Cal.com users either way! ❤️

**Comment by erulabs** [comment]
Ah, I see what you mean about the `web_app_url` now, as in `NEXT_PUBLIC_WEBAPP_URL` is set during image build time.

I suppose I could run `yarn build` again, or possibly tell Turbo to re-build the front-end for me? I am familiar with Turbo, but I see https://github.com/calcom/cal.com/blob/6b0ac96b38b0dbd78809a73e19010192f31cc769/turbo.json#L29-L43 references depending on an environment variable. I've supplied the correct NEXT_PUBLIC_WEBAPP_URL but it doesn't seem to rebuild any assets on start.

Here is a link to the Kubernetes YAML so far for anyone looking to deploy Cal.com on Kubernetes! https://kubesail.com/template/erulabs/Cal.com

**Comment by DrMxrcy** [comment]
Yea I see the localhost issue. I have been working around
 it so far!

**Comment by krumware** [comment]
@DrMxrcy @erulabs official version merged and incoming under `:latest`
@erulabs that's awesome about kubesail. Is there anything we should consider coordinating there when the official helm deployment comes out?

**Comment by erulabs** [comment]
Heya @krumware - we support helm charts too! 👯 I'll update our non-helm manifests and see if I cant get things working. By the way - if you're interested in collaborating, we do have a revenue share agreement! I'd love to sell PiBoxes preloaded with Cal.com to doctors offices (for example!). Let me know if that sparks any ideas for ya! I'll reach out to the proper channels once I get the app working properly :)

**Comment by DrMxrcy** [comment]
@krumware Thanks so much for all the work on the docker image! Can't wait for it to build so we can test it out!

**Comment by Xosten** [comment]
Hey @krumware,

I'm commenting on this issue as I'm trying to run a docker image on a OCI ARM Free Tier (Ampere). I'm getting this error each time and wonder if the image is compatible with `aarch64` ? 

`````
@calcom/web:build: info  - Linting and checking validity of types...
@calcom/web:build: warn  - No ESLint configuration detected. Run next lint to begin setup
@calcom/web:build: info  - Creating an optimized production build...
@calcom/web:build: <jemalloc>: Unsupported system page size
@calcom/web:build: <jemalloc>: Unsupported system page size
@calcom/web:build: memory allocation of 10 bytes failed
@calcom/web:build: Aborted (core dumped)
@calcom/web:build: error Command failed with exit code 134.
@calcom/web:build: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
@calcom/web:build: ERROR: command finished with error: command (apps/web) yarn run build exited (134)
command (apps/web) yarn run build exited (134)

 Tasks:    7 successful, 8 total
Cached:    0 cached, 8 total
  Time:    2m2.068s 

error Command failed with exit code 134.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
1 error occurred:
        * Status: The command '/bin/sh -c yarn build' returned a non-zero code: 134, Code: 134
````

**Comment by krumware** [comment]
Thanks for the heads up, I'll work to add arm7 which should cover it

**Comment by Xosten** [comment]
> Thanks for the heads up, I'll work to add arm7 which should cover it

Awesome! Can't wait to try it out!

**Comment by krumware** [comment]
Checking now, but not sure, are you trying to build the image yourself on aarch64, or is this while pulling then running the latest docker image on aarch64?

**Comment by Xosten** [comment]
> Checking now, but not sure, are you trying to build the image yourself on aarch64, or is this while pulling then running the latest docker image on aarch64?

This is while running the build command `DOCKER_BUILDKIT=0 docker compose build calcom`

**Comment by krumware** [comment]
it's not the image itself then. if you want to test the image on aarch64 you should be able to pull the latest image and run it.

we probably need a new issue to track against.
honestly, it looks like your build runner or the machine itself ran out of memory during the build, or it is some other issue. here's a similar issue on home assistant https://github.com/home-assistant/core/issues/34162

**Comment by Xosten** [comment]
> it's not the image itself then. if you want to test the image on aarch64 you should be able to pull the latest image and run it.
> 
> we probably need a new issue to track against. honestly, it looks like your build runner or the machine itself ran out of memory during the build, or it is some other issue. here's a similar issue on home assistant [home-assistant/core#34162](https://github.com/home-assistant/core/issues/34162)

I will test by pulling the latest image then, but it's meant for testing?

I doubt that my machine ran out of memory as it has 24Gb or Ram, 4vCPUs and approx 50/100Gb free storage. I'm not running any heavy process currently on it. Is there maybe an OS conflicting? I'm running Oracle Linux 8.

I'll open a seperate issue regarding this.

**Comment by krumware** [comment]
Correct, it's currently meant for testing but production images are being worked on. If you can pull the image, there's already an arch variant built for aarch64 that should run. If it doesn't run, then there might be other issues to track, so your running it will be helpful intel.

Correct, it's likely something environment related during the build, but we'll figure it out together. Thanks for the OS/machine details, it's good insight!

**Comment by Xosten** [comment]
> Correct, it's currently meant for testing but production images are being worked on. If you can pull the image, there's already an arch variant built for aarch64 that should run. If it doesn't run, then there might be other issues to track, so your running it will be helpful intel.
> 
> Correct, it's likely something environment related during the build, but we'll figure it out together. Thanks for the OS/machine details, it's good insight!

Pulling the latest image worked. I have opened #156 to keep track of this. Let's move over there the conversation :)

**Comment by joshavant** [comment]
+1-ing this request
